src/mono/wasm/runtime/startup.ts

```diff
@@ -557,9 +562,25 @@ async function mono_wasm_before_memory_snapshot() {
         runtimeHelpers.storeMemorySnapshotPending = false;
     }
 
+    if (runtimeHelpers.config.interpreterPgo) {
+        if (typeof (globalThis.setTimeout) === "function")
+            globalThis.setTimeout(maybeSaveInterpPgoTable, (runtimeHelpers.config.interpreterPgoSaveDelay || 15) * 1000);
+    }
+
     endMeasure(mark, MeasuredBlock.memorySnapshot);
 }
 
+async function maybeSaveInterpPgoTable () {
+    // If the application exited abnormally, don't save the table. It probably doesn't contain useful data,
+    //  and saving would overwrite any existing table from a previous successful run.
+    // We treat exiting with a code of 0 as equivalent to if the app is still running - it's perfectly fine
+    //  to save the table once main has returned, since the table can still make future runs faster.
+    if ((loaderHelpers.exitCode !== undefined) && (loaderHelpers.exitCode !== 0))
```

Comment:
web apps never exit

Reply:
I don't understand, wasn't this your suggestion?

Reply:
my bad, I misread the logic. 
It means take snapshot if it's still running or if it exited with zero, right ?

Reply:
Correct, we don't want to save a PGO table for a run that asserted or failed